parachain/primitives/core/src/outbound.rs

```diff
@@ -129,18 +134,48 @@ impl Command {
 		}
 	}
 
+	/// Compute gas cost
+	/// Todo: load by trait from configurable, reference gas from benchmark report
+	/// with some extra margin
+	/// | Function Name    | min    | avg    | median | max    | #
+	/// calls | | createAgent    | 839    | 184709 | 237187 | 237187 | 9       |     
+	/// | createChannel    | 399    | 31023  | 2829   | 75402  | 5       |
+	/// | updateChannel    | 817    | 15121  | 3552   | 36762  | 5     |     
+	/// | transferNativeFromAgent    | 770    | 21730  | 21730  | 42691  | 2       |
+	pub fn dispatch_gas(&self) -> u128 {
+		match self {
+			Command::AgentExecute { .. } => 500000,
+			Command::CreateAgent { .. } => 250000,
+			Command::CreateChannel { .. } => 90000,
+			Command::UpdateChannel { .. } => 50000,
+			Command::TransferNativeFromAgent { .. } => 60000,
+			// For sudo operations set as zero do not charge fees
+			Command::SetOperatingMode { .. } => 0,
+			Command::Upgrade { .. } => 0,
+		}
+	}
+
+	// Todo: configurable gas price, for now set as 20 Gwei
+	// TBC: 4/5 to relayer and 1/5 left for protocol maintenance
+	pub fn reward(&self) -> u128 {
+		let gas_price: u128 = 20_000_000_000;
+		self.dispatch_gas() * gas_price * 4 / 5
+	}
```

Comment:
Make `dispatch_gas` and  `reward` inherent attribute of the command. The `reward` is linear correlation with the gas cost.